.github/workflows/odh-notebooks-sync.yml

```
@@ -0,0 +1,84 @@
+# The aim of this GitHub workflow is to update the pyproject.toml to sync with ODH Jupyter Trash Cleanup release.
+name: Sync ODH-notebooks with ODH Jupyter Trash Cleanup release
+on:
+  workflow_dispatch:
+    inputs:
+      upstream-repository-organization:
+        required: true
+        description: "Owner of target upstream notebooks repository used to open a PR against"
+        default: "opendatahub-io"
+      notebooks-target-branch:
+        required: true
+        description: "Target branch of upstream repository"
+        default: "main"
+      repository-organization:
+        required: true
+        description: "Owner of origin notebooks repository used to open a PR"
+        default: "ide-developer"
+      repository-name:
+        required: true
+        description: "Name of the target repository"
+        default: "notebooks"
+      odh_jupyter_trash_cleanup_release_version:
+        required: true
+        description: "Provide version of the ODH Jupyter Trash Cleanup release"
+
+env:
+  BRANCH_NAME: ${{ github.event.inputs.notebooks-target-branch }}
+  ODH_JUPYTER_TRASH_CLEANUP_RELEASE_VERSION: ${{ github.event.inputs.odh_jupyter_trash_cleanup_release_version }}
+  UPDATER_BRANCH: odh-sync-updater-${{ github.run_id }}
+  UPSTREAM_OWNER: ${{ github.event.inputs.upstream-repository-organization }}
+  REPO_OWNER: ${{ github.event.inputs.repository-organization }}
+  REPO_NAME: ${{ github.event.inputs.repository-name }}
+  GITHUB_TOKEN: ${{ secrets.IDE_DEVELOPER_TOKEN }}
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Clone repository and Sync
+        run: |
+          git clone https://x-access-token:${GITHUB_TOKEN}@github.com/$REPO_OWNER/$REPO_NAME.git $REPO_NAME
+          cd $REPO_NAME
+          git remote add upstream https://github.com/$UPSTREAM_OWNER/$REPO_NAME.git
+          git config --global user.email "ide-developer@users.noreply.github.com"
+          git config --global user.name "ide-developer"
+          git checkout $BRANCH_NAME
+          git pull --ff-only upstream $BRANCH_NAME && git push origin $BRANCH_NAME
+
+      - name: Update pyproject.toml with Trash Cleanup release version
+        run: |
+          cd "$REPO_NAME"
+          find . -name "pyproject.toml" | while read file; do
+            sed -i "s|odh-jupyter-trash-cleanup = \".*\"|odh-jupyter-trash-cleanup = \"~=${ODH_JUPYTER_TRASH_CLEANUP_RELEASE_VERSION}\"|" "$file"
+          done
+          if ! git diff --quiet; then
+            make refresh-pipfilelock-files
+          fi
```

Since we expect the update of the extension to happen, maybe we should end with error if there are no changes?
```suggestion
          else
            echo "No changes detected; this is not expected!"
            exit 1
          fi
```